src/test/unit/cli_unittest.cc

```diff
@@ -205,6 +222,283 @@ TEST(CLIUnittest, TestCliSetStringWriteOnce)
     EXPECT_EQ(0,   data[6]);
 }
 
+static uint8_t data[1000];
+static vector<string> outLines;
+
+class CliWriteTest : public ::testing::Test
+{
+
+protected:
+    static void SetUpTestCase() {}
+
+    virtual void SetUp() {
+        bufWriterInit(&cliWriterDesc, data, ARRAYLEN(data), &dummyBufWriter, NULL);
+        cliWriter = cliErrorWriter = &cliWriterDesc;
+    }
+
+    virtual void TearDown() {
+        cliWriter = NULL;
+        outLines.clear();
+    }
+
+    static void dummyBufWriter(void *, void *_data, int size) {
+        uint8_t *data = (uint8_t *)_data;
+        ostringstream stream;
+        for (int i = 0; i < size; i++) {
+            stream << (char)data[i];
+        }
+        outLines.push_back(stream.str());
+    }
+
+    void clear() {
+        outLines.clear();
+    }
+};
+
+class AuxCliWriteTest : public CliWriteTest {
+protected:
+    virtual void SetUp() {
+        CliWriteTest::SetUp();
+
+        for (uint8_t i = 0; i <= MAX_AUX_CHANNEL_COUNT; i++) {
```

Comment:
(unless you want to explicitly assume case when MAX_AUX_CHANNEL_COUNT >=256)
```suggestion
        for (unsigned i = 0; i <= MAX_AUX_CHANNEL_COUNT; i++) {
```